charset_normalizer/probe_chaos.py

```diff
@@ -68,60 +71,17 @@ def __init__(self, string, giveup_threshold=0.09, bonus_bom_sig=False, bonus_mul
 
         self._probe()
 
-    def __add__(self, other):
-        """
-        :param ProbeChaos other:
-        :return:
-        """
-        k_ = ProbeChaos('', self._threshold)
-
-        k_.successive_upper_lower = self.successive_upper_lower + other.successive_upper_lower
-        k_.successive_accent = self.successive_accent + other.successive_accent
-        k_.successive_different_unicode_range = self.successive_different_unicode_range + other.successive_different_unicode_range
-
-        for el in self.encountered_unicode_range:
-            k_.encountered_unicode_range.add(el)
-
-        for el in other.encountered_unicode_range:
-            k_.encountered_unicode_range.add(el)
-
-        k_.encountered_punc_sign = self.encountered_punc_sign + other.encountered_punc_sign
-        k_.unprintable = self.unprintable + other.unprintable
-        k_.encountered_white_space = self.encountered_white_space + other.encountered_white_space
-        k_.not_encountered_white_space = self.not_encountered_white_space + other.not_encountered_white_space
-
-        for u_name, u_occ in self.encountered_unicode_range_occurrences.items():
-            if u_name not in k_.encountered_unicode_range_occurrences.keys():
-                k_.encountered_unicode_range_occurrences[u_name] = 0
-            k_.encountered_unicode_range_occurrences[u_name] += u_occ
-
-        for u_name, u_occ in other.encountered_unicode_range_occurrences.items():
-            if u_name not in k_.encountered_unicode_range_occurrences.keys():
-                k_.encountered_unicode_range_occurrences[u_name] = 0
-            k_.encountered_unicode_range_occurrences[u_name] += u_occ
-
-        k_.not_encountered_white_space_reset = self.not_encountered_white_space_reset + other.not_encountered_white_space_reset
-        k_.total_letter_encountered = self.total_letter_encountered + other.total_letter_encountered
-        k_.total_lower_letter_encountered = self.total_lower_letter_encountered + other.total_lower_letter_encountered
-        k_.total_upper_accent_encountered = self.total_upper_accent_encountered + other.total_upper_accent_encountered
-        k_.total_upper_accent_encountered_inner = self.total_upper_accent_encountered_inner + other.total_upper_accent_encountered_inner
-        k_.total_unaccented_letter_encountered = self.total_unaccented_letter_encountered + other.total_unaccented_letter_encountered
-
-        k_._probe_word = self._probe_word + other._probe_word
-
-        k_._string = self._string + other._string
-
-        return k_
-
     def _probe(self):
 
         c__ = False
         upper_lower_m = False
 
-        for c, i_ in zip(self._string, range(0, len(self._string))):
+        str_len = len(self._string)
+
+        for c, i_ in zip(self._string, range(0, str_len)):

```

Comment:
one character  variable is not pythonic.

Reply:
Use `char` and `index`

Reply:
I know that, that because `for c, i_ in zip(self._string, range(0, str_len)):` is faster than `for index in range(0, str_len): # then pick char by index` for some reason.

Reply:
will do. there is some other place too.